src/sidebar/helpers/annotation-user.js

```diff
@@ -17,18 +18,59 @@ import { username } from './account-id';
  * username or the display name.
  *
  * @param {Pick<Annotation, 'user'|'user_info'>} annotation
- * @param {boolean} isThirdPartyUser - Is the annotation's user third-party?
- * @param {boolean} isFeatureFlagEnabled - Is the `client_display_names`
- *   feature flag enabled
+ * @param {SidebarStore} store
+ *
  * @return {string}
  */
-export function annotationDisplayName(
-  annotation,
-  isThirdPartyUser,
-  isFeatureFlagEnabled
-) {
-  const useDisplayName = isFeatureFlagEnabled || isThirdPartyUser;
+export function annotationDisplayName(annotation, store) {
+  const defaultAuthority = store.defaultAuthority();
+  const isThirdParty = isThirdPartyUser(annotation.user, defaultAuthority);
+
+  const displayNamesEnabled = store.isFeatureEnabled('client_display_names');
+
+  const useDisplayName = displayNamesEnabled || isThirdParty;
   return useDisplayName && annotation.user_info?.display_name
     ? annotation.user_info.display_name
     : username(annotation.user);
 }
+
+/**
+ * Return a URL to the annotation author's user page, when available. Author
+ * links for third-party users are only available if a `usernameUrl` is
+ * provided in `settings`.
+ *
+ * @param {Pick<Annotation, 'user'>} annotation
+ * @param {SidebarStore} store
+ * @param {SidebarSettings} settings
+ */
+export function annotationAuthorLink(annotation, store, settings) {
+  const defaultAuthority = store.defaultAuthority();
+  const isThirdParty = isThirdPartyUser(annotation.user, defaultAuthority);
+
+  const authorLink = (() => {
+    if (!isThirdParty) {
+      return store.getLink('user', { user: annotation.user });
+    } else {
+      return (
+        (settings.usernameUrl &&
+          `${settings.usernameUrl}${username(annotation.user)}`) ??
+        undefined
+      );
+    }
+  })();
+  return authorLink;
```

Comment:
This IIFE seems redundant, the `authorLink` variable is not used internally after being created. In other words, you could replace it with the content of the function and get the same result.